remote-vfs.c

```diff
@@ -0,0 +1,1889 @@
+// TODO Write a man page.  Here are some notes for dogfooding.
+// TODO
+//
+// Usage: git remote-vfs [<main_options>] <sub-command> [<sub-command-options>]
+//
+// <main_options>:
+//
+//     --remote=<remote-name>         // defaults to "origin"
+//
+//     --mode=<mode>                  // defaults to "auto"
+//
+//            auto    := auto-detect whether repo is VFS4G or Scalar mode.
+//            vfs     := force VFS4G mode.
+//            scalar  := force Scalar mode.
+//            none    := assume neither mode.
+//
+//            The <mode> is used to look for mode-specific config settings:
+//            "<mode>.cache-server" and "core.<mode>".
```

Comment:
Note: at the moment, we still use `core.gvfs`. This may change in the future, so having the comment here is helpful.